packages/browser/src/integrations/breadcrumbs.ts

```diff
@@ -313,18 +313,38 @@ export class Breadcrumbs implements Integration {
 function addSentryBreadcrumb(serializedData: string): void {
   // There's always something that can go wrong with deserialization...
   try {
-    const event = JSON.parse(serializedData);
-    getCurrentHub().addBreadcrumb(
-      {
-        category: `sentry.${event.type === 'transaction' ? 'transaction' : 'event'}`,
-        event_id: event.event_id,
-        level: event.level || Severity.fromString('error'),
-        message: getEventDescription(event),
-      },
-      {
-        event,
-      },
-    );
+    try {
+      const event = JSON.parse(serializedData);
+      getCurrentHub().addBreadcrumb(
+        {
+          category: `sentry.${event.type === 'transaction' ? 'transaction' : 'event'}`,
+          event_id: event.event_id,
+          level: event.level || Severity.fromString('error'),
+          message: getEventDescription(event),
+        },
+        {
+          event,
+        },
+      );
+    } catch (_oO) {
+      // We are dealing with an envelope here
+      // For simplicity we only deal with transactions
+      const envelopeLines = serializedData.split('\n');
```

Comment:
@HazAT  Is it guaranteed that the none of the entries will contain newlines?

Reply:
Yes, per envelope spec.

Reply:
@HazAT 
I read over the spec. There's a chance the item payload could contain newlines; and this could fail:

```js
const item = JSON.parse(envelopeLines[2]);
```

Maybe something like this?

```js
const [envelopeHeaderRaw, itemHeaderRaw, ...rest] = serializedData.split('\n');

const envelopeHeader = JSON.parse(envelopeHeaderRaw);
const itemHeader = JSON.parse(itemHeaderRaw);
const body = rest.join('\n');
```